components/TextFields/examples/TextFieldExample.swift

```diff
@@ -0,0 +1,749 @@
+
+    textFieldLeftView.placeholder = "This has a left view"
+    textFieldLeftView.delegate = self
+    textFieldLeftView.clearButtonMode = .whileEditing
+
+    let textFieldControllerDefaultLeftView = MDCTextInputController(input: textFieldLeftView)
+
+    let textFieldLeftViewFloating = MDCTextField()
+    textFieldLeftViewFloating.leftViewMode = .always
+    textFieldLeftViewFloating.leftView = UIImageView(image:leftViewImage)
+
+    scrollView.addSubview(textFieldLeftViewFloating)
+    textFieldLeftViewFloating.translatesAutoresizingMaskIntoConstraints = false
+
+    textFieldLeftViewFloating.placeholder = "This has a left view and floats"
+    textFieldLeftViewFloating.delegate = self
+    textFieldLeftViewFloating.clearButtonMode = .whileEditing
+
+    let textFieldControllerDefaultLeftViewFloating =
+      MDCTextInputController(input: textFieldLeftViewFloating)
+    textFieldControllerDefaultLeftViewFloating.presentation = .floatingPlaceholder
+
+    let rightViewImagePath = bundle.path(forResource: "ic_done", ofType: "png")!
+    let rightViewImage = UIImage(contentsOfFile: rightViewImagePath)!
+
+    let textFieldRightView = MDCTextField()
+    textFieldRightView.rightViewMode = .always
+    textFieldRightView.rightView = UIImageView(image:rightViewImage)
+
+    scrollView.addSubview(textFieldRightView)
+    textFieldRightView.translatesAutoresizingMaskIntoConstraints = false
+
+    textFieldRightView.placeholder = "This has a right view"
+    textFieldRightView.delegate = self
+    textFieldRightView.clearButtonMode = .whileEditing
+
+    let textFieldControllerDefaultRightView = MDCTextInputController(input: textFieldRightView)
+
+    let textFieldRightViewFloating = MDCTextField()
+    textFieldRightViewFloating.rightViewMode = .always
+    textFieldRightViewFloating.rightView = UIImageView(image:rightViewImage)
+
+    scrollView.addSubview(textFieldRightViewFloating)
+    textFieldRightViewFloating.translatesAutoresizingMaskIntoConstraints = false
+
+    textFieldRightViewFloating.placeholder = "This has a right view and floats"
+    textFieldRightViewFloating.delegate = self
+    textFieldRightViewFloating.clearButtonMode = .whileEditing
+
+    let textFieldControllerDefaultRightViewFloating =
+      MDCTextInputController(input: textFieldRightViewFloating)
+    textFieldControllerDefaultRightViewFloating.presentation = .floatingPlaceholder
+
+    let textFieldLeftRightView = MDCTextField()
+    textFieldLeftRightView.leftViewMode = .whileEditing
+    textFieldLeftRightView.leftView = UIImageView(image: leftViewImage)
+    textFieldLeftRightView.rightViewMode = .unlessEditing
+    textFieldLeftRightView.rightView = UIImageView(image:rightViewImage)
+
+    scrollView.addSubview(textFieldLeftRightView)
+    textFieldLeftRightView.translatesAutoresizingMaskIntoConstraints = false
+
+    textFieldLeftRightView.placeholder =
+      "This has left & right views and a very long placeholder that should be truncated"
```

Comment:
This text field doesn't appear to be being truncated until i give it focus and then remove focus.